WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchSegmentsUseCase.kt

```diff
@@ -0,0 +1,39 @@
+package org.wordpress.android.ui.sitecreation.usecases
+
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.VerticalActionBuilder
+import org.wordpress.android.fluxc.store.VerticalStore.OnSegmentsFetched
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+import kotlin.coroutines.experimental.suspendCoroutine
+
+/**
+ * Transforms EventBus event to a coroutines.
+ */
+class FetchSegmentsUseCase @Inject constructor(val dispatcher: Dispatcher) {
+    private var continuation: Continuation<OnSegmentsFetched>? = null
+
+    suspend fun fetchCategories(): OnSegmentsFetched {
+        if (continuation != null) {
+            throw IllegalStateException("Fetch already in progress.")
+        }
+        return suspendCoroutine { cont ->
+            continuation = cont
+            dispatchEvent()
+        }
+    }
+
+    private fun dispatchEvent() {
```

Comment:
:D good question - during the first iteration the method contained more than just a single line of code as I was mocking the whole FluxC. It would still make sense in some cases to have a method with a single line to make the code more readable -> however, in this case since the methods is named `dispatchEvent` and it calls `dispatcher.dispatch`, it doesn't make any sense at all. I'll remove it.